src/cldr.js

```diff
@@ -140,85 +303,74 @@ export function createDateTimeFormats(formats) {
+            computed = createDateTimeFormat(skeleton, pattern);
             if (computed) {
                 result.push(computed);
                 // in some cases, the format is only displaying date specific props
                 // or time specific props, in which case we need to also produce the
                 // combined formats.
                 if (isDateFormatOnly(computed)) {
-                    dateRelatedFormats.push(format);
+                    dateRelatedFormats.push(computed);
                 } else if (isTimeFormatOnly(computed)) {
-                    timeRelatedFormats.push(format);
+                    timeRelatedFormats.push(computed);
                 }
             }
         }
     }
 
-    // combine custom time and custom date formats when they are orthogonals to complete the
-    // formats supported by browsers by relying on the value of "formats.medium" which defines
-    // how to join custom formats into a single pattern.
-    for (i = 0; i < timeRelatedFormats.length; i += 1) {
-        for (j = 0; j < dateRelatedFormats.length; j += 1) {
-            format = order
-                .replace('{0}', timeRelatedFormats[i])
-                .replace('{1}', dateRelatedFormats[j])
-                .replace(/^[,\s]+|[,\s]+$/gi, '');
-            computed = createDateTimeFormat(format);
+    // Map time formats into a pattern for createDateTimeFormats
+    for (skeleton in timeFormats) {
+        if (timeFormats.hasOwnProperty(skeleton)) {
+            pattern = timeFormats[skeleton];
+            computed = createDateTimeFormat(skeleton, pattern);
             if (computed) {
                 result.push(computed);
+                timeRelatedFormats.push(computed);
             }
         }
     }
 
-    // Map time formats into a pattern for createDateTimeFormats
-    for (key in timeFormats) {
-        if (timeFormats.hasOwnProperty(key)) {
-            format = expandFormat(key, timeFormats[key]);
-            computed = createDateTimeFormat(format);
+    // Map date formats into a pattern for createDateTimeFormats
+    for (skeleton in dateFormats) {
+        if (dateFormats.hasOwnProperty(skeleton)) {
+            pattern = dateFormats[skeleton];
+            computed = createDateTimeFormat(skeleton, pattern);
             if (computed) {
                 result.push(computed);
+                dateRelatedFormats.push(computed);
             }
         }
     }
 
-    // Map date formats into a pattern for createDateTimeFormats
-    for (key in dateFormats) {
-        if (dateFormats.hasOwnProperty(key)) {
-            format = expandFormat(key, dateFormats[key]);
-            computed = createDateTimeFormat(format);
-            if (computed) {
-                result.push(computed);
+    // combine custom time and custom date formats when they are orthogonals to complete the
+    // formats supported by CLDR.
+    // This Algo is based on section "Missing Skeleton Fields" from:
```

Comment:
date and time will be mixed together according to the algo described in Missing Skeleton Fields.